### PR TITLE
Fix: correct install routine when activating in a WordPress multisite.

### PR DESCRIPTION
**Pull request by pauloiankoski**
## Description
This PR addresses an error in the installation process of GiveWP on multisite setups. Previously, only the main site was installed, resulting in fatal errors on existing subsites.

Upon investigating the codebase, it was discovered that the installation method for GiveWP had a conditional check for `is_multisite()` and a `$network_wide` parameter. However, this parameter was never set to `true`, causing improper installation on WP multisite.

In this PR, we enable the parameter when the plugin is activated for the first time, ensuring proper installation of GiveWP on all existing subsites.

## Affects

GiveWP install process

## Testing Instructions

1. Fresh WordPress install with Multiste enabled
2. Create one subsite at My Sites > Network Admin > Add New
3. Navigate to My Sites > Network Admin > Plugins, install GiveWP, and activate it
4. Navigate to My Sites > Network Admin > Sites, scroll to subsite, click "Dashboard" link
5. The subsite dashboard must show and the Donations menu must be present, meaning that Give is activated

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit tests
-   [ ] Reviewed by the designer (if follows a design)
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205607759526942

### CODE REVIEW

File: give.php [hunk 1]
```diff
@@ -404,10 +404,18 @@ public function display_old_recurring_compatibility_notice()
...
+     * Runs on plugin activation and performs initial setup.
+     *
+     * @unreleased set network_wide parameter to true, enabling installing in WP multisite
+     * @since 1.0.0
+     */
     public function install()
     {
         $this->loadServiceProviders();
-        give_install();
+        give_install(true);
```
Comment (JasonTheAdams) [inline give.php:418]: @pauloiankoski Why even have the parameter if it's going to be fixed as true? I can see the function is already checking for `is_multisite()`, so I'm not sure what more value the parameter provides.

Comment (pauloiankoski) [reply give.php:418]: That function is called in other places like when creating a new subsite. So, even tho, it's a multisite, we don't want to reinstall Give across all subsites, but only in the new one.

CleanShot 2023-09-28 at 18 40 31

Comment (JasonTheAdams) [reply give.php:418]: Ahhh, I see. Good catch. 👍

**Review by JasonTheAdams — APPROVED** [review]
Thanks for this, @pauloiankoski! Let's make sure QA tests different types of sites — single, multi, and new sub-site installs.

**Comment by Genevieve-K** [comment]
I don't understand this commit. When I activate the plugin on a sub-site, I want it only on this sub-site.
When we activate it on the root site as network enabled, I understand that the plugin is installed on all sites. 
And I have a fatal error: 
`Fatal error: Uncaught TypeError: array_key_exists(): Argument #2 ($array) must be of type array, null given in /var/www/html/wordpress/wp-content/plugins/give/src/PaymentGateways/Gateways/PayPalStandard/Migrations/SetPayPalStandardGatewayId.php:30 `

The support suggests me to delete the give settings and the wp_give tables for the root site. It works only if I changed this $network_wide parameter to false.

**Pull request merged**